api/document/js/__tests__/commands.test.ts

```diff
@@ -187,9 +191,59 @@ describe('makeSaveThenXml()', () => {
 
     const api = new JsonApi({ csrfToken: '', url: '' });
     const save = makeSaveThenXml(api);
-    await save({ doc: 'stuff' });
+    await save(EditorState.create({ doc: factory.policy() }));
 
     const param = locationAssign.mock.calls[0][0];
     expect(param).toMatch(/akn$/);
   });
 });
+
+describe('addListItem()', () => {
+  const doc = factory.policy([
+    factory.para('intro'),
+    factory.list([
+      factory.listitem('a:', [factory.para('aaa')]),
+      factory.listitem('b:', [
+        factory.para('bbb first'),
+        factory.para('bbb second'),
+      ]),
+    ]),
+  ]);
+
+  function makeState(path: SelectionPath) {
+    const selection = new TextSelection(pathToResolvedPos(doc, path));
+    return EditorState.create({ doc, selection });
+  }
+
+  it('requires the cursor be in the right position', () => {
+    const inIntro = makeState(['para', 'inline', 'int'.length]);
+    const endIntro = makeState(['para', 'inline', 'intro'.length]);
+    const middleOfLi = makeState(['list', 'listitem', 'para', 'inline', 'a'.length]);
+    const endOfFirstPara = makeState(
+      ['list', new NthType(1, 'listitem'), 'para', 'inline', 'bbb first'.length]);
+    [inIntro, endIntro, middleOfLi, endOfFirstPara].forEach(state =>
+      expect(addListItem(state)).toBe(false));
```

Comment:
This reminded me of py.test's concept of "parameterized tests" and after a bit of googling it turned up [`jest-each`](https://www.npmjs.com/package/jest-each).  If we think we'll be writing tests like this often, it might be worth adding later, outside of this PR.

Reply:
Good idea; I'll open a ticket.